### PR TITLE
Don't release uncompressed binaries under the new naming scheme

### PR DESCRIPTION
**Pull request by lnicola**
bors r+

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -82,7 +82,7 @@ jobs:
 
     - name: Nightly analysis-stats check
       if: github.ref != 'refs/heads/release'
-      run: ./dist/rust-analyzer-x86_64-unknown-linux-gnu analysis-stats .
+      run: target/x86_64-unknown-linux-gnu/release/rust-analyzer analysis-stats .
```
Comment (matklad) [inline .github/workflows/release.yaml:85]: Couple of drive-bys:

* the `./dist` was intentional here, to make sure that the final thing works. Arguably, the stats job should even download it from GitHub releases page... But this is minor, really
* Let's set `RA_TARGET` env var on the job level, if possible?

Comment (lnicola) [reply .github/workflows/release.yaml:85]: > the ./dist was intentional here, to make sure that the final thing works. Arguably, the stats job should even download it from GitHub releases page... But this is minor, really

Yeah, but that file no longer exists (since the binary is compressed). We could `gunzip` it, but I'm not sure it matters too much.

> Let's set RA_TARGET env var on the job level, if possible?

Sure

Comment (matklad) [reply .github/workflows/release.yaml:85]: ahhh, right, yeah, makes sense!

Comment (lnicola) [reply .github/workflows/release.yaml:85]: > ahhh, right, yeah, makes sense!

Ask me how I found out :smile:
